components/serialization/form/src/test/java/com/microsoft/kiota/serialization/mocks/TestEntity.java

```diff
@@ -54,8 +54,8 @@ public PeriodAndDuration getWorkDuration() {
 		return _workDuration;
 	}
 
-	public void setWorkDuration(PeriodAndDuration value) {
-		this._workDuration = value;
+	public void setWorkDuration(final PeriodAndDuration value) {
```

Comment:
Just to confirm here, if this change is a side effect of removing the immutable annotation, will we need a corresponding change in the generation logic to avoid a warning/error?

Reply:
As far as I understand it, this change is not needed.

Reply:
The 'final' is not needed, however the way that the workDuration values is set
``` Java 
this._workDuration = PeriodAndDuration.of(value.getPeriod(), value.getDuration());
```
Will likely need to change in generation. 
@baywet , @andrueastman , do we feel this is a fair pattern to follow in generation?

Reply:
@ramsessanchez This looks good to me. I understand the need to do this as the removal of the `Immutable` annotation leads to the analyzers causing build error.